### PR TITLE
Remove/raise exception when external file exists during onnx.save

### PR DESCRIPTION
**Pull request by tonypottera24**
### Description

Raise an exception when the external file exists during `onnx.save`.

Note: we can also consider to overwrite it.

### Motivation and Context

If the external file is not removed, it will not be overwritten. The new content is always appended to it, making it larger and larger forever.

### CODE REVIEW

**Comment by xadupre** [comment]
Should we remove it or raise an exception?

**Comment by tonypottera24** [comment]
> Should we remove it or raise an exception? cc @gramalingam

Should I do this here or inside `onnx.save`?

**Comment by justinchuby** [comment]
I think we should overwrite, but I am not sure if that is entirely robust either: I can imagine when part of the tensors are external in the model pointing to `model.data`, after calling save(), if we overwrite, the original external data would be gone so the original references will be invalid and get undetected (silent incorrectness).

However if we raise, the user will then need to manually remove the data file. This is `O(# tensors)` and quite burdensome in the case when `all_tensors_to_one_file` is false because there would be many files.

The current behavior of appending is actually the only safe way I can think of, but admittedly not optimal as explained in this PR.

**Review by justinchuby — APPROVED** [review]
LGTM thanks. Please ignore my previous comments

**Comment by justinchuby** [comment]
Could you add the `Raises` section in the docstring to describe this new behavior. Thanks! ref: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Comment by justinchuby** [comment]
Overwriting can be unsafe if the model proto has reference to the existing data file. Maybe we can provide an option? But I also don't prefer overcrowding the options we have. hmm

**Review by justinchuby — CHANGES_REQUESTED** [review]
Deferring to @gramalingam

**Comment by tonypottera24** [comment]
> Could you add the `Raises` section in the docstring to describe this new behavior. Thanks! ref: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

Thank you @justinchuby, I added the explanations.

**Comment by tonypottera24** [comment]
I saw `offset` in function `save_external_data`...

**Comment by gramalingam** [comment]
I think it is safer not to overwrite it silently. I am fine with using a user option to overwrite or leaving in the PR's current form (with an exception).

File: onnx/external_data_helper.py [hunk 1]
```diff
@@ -127,18 +127,24 @@ def convert_model_to_external_data(
...
     if all_tensors_to_one_file:
-        file_name = str(uuid.uuid1())
+        file_name = str(uuid.uuid1()) + ".data"
         if location:
             if os.path.isabs(location):
                 raise ValueError(
                     "location must be a relative path that is relative to the model path."
                 )
+            if os.path.exists(location):
+                raise FileExistsError(f"File exists in {location}.")
```
Comment (justinchuby) [inline onnx/external_data_helper.py:147]:
```suggestion
                raise FileExistsError(f"External file exists in {location}.")
```

File: onnx/external_data_helper.py [hunk 2]
```diff
@@ -127,18 +127,24 @@ def convert_model_to_external_data(
...
     if all_tensors_to_one_file:
-        file_name = str(uuid.uuid1())
+        file_name = str(uuid.uuid1()) + ".data"
         if location:
             if os.path.isabs(location):
                 raise ValueError(
                     "location must be a relative path that is relative to the model path."
                 )
+            if os.path.exists(location):
+                raise FileExistsError(f"External file exists in {location}.")
```
Comment (justinchuby) [inline onnx/external_data_helper.py:147]:
```suggestion
                raise FileExistsError(f"External data file exists in {location}.")
```

**Review by justinchuby — APPROVED** [review]

**Pull request merged**